### PR TITLE
Implement unit tests for NotificationService

### PR DESCRIPTION
**Pull request by moltenessence**

### CODE REVIEW

**Review by nantonov — APPROVED** [review]

File: PMCS/Services/NotificationService/Notification.Tests/AutoData/AutoNotificationDataAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+using AutoFixture.AutoMoq;
+using AutoFixture.Xunit2;
+using Notifications.API.Models.Requests;
+
+namespace BLL.Tests.AutoData
+{
+    public class AutoNotificationDataAttribute : AutoDataAttribute
+    {
+        private const string testAddress = "test@gmail.com";
+        private const string testUserId = "1";
```
Comment (Blackolf931) [inline PMCS/Services/NotificationService/Notification.Tests/AutoData/AutoNotificationDataAttribute.cs:11]: move to a separate file

**Review by Blackolf931 — CHANGES_REQUESTED** [review]

File: PMCS/Services/NotificationService/Notification.Tests/UnitTests/NotificationTests.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+namespace BLL.Tests.UnitTests
+{
+    public class NotificationTests
+    {
+        [Theory, AutoNotificationData]
+        public async Task NotifyByEmail_ValidRequest_SendsEmail(
+            EmailNotificationRequest request,
+            EmailNotification notification,
+            [Frozen] Mock<IEmailService> emailServiceMock,
```
Comment (Blackolf931) [inline PMCS/Services/NotificationService/Notification.Tests/UnitTests/NotificationTests.cs:19]: Please give me a link to read about [Frozen] and etc

Comment (moltenessence) [reply PMCS/Services/NotificationService/Notification.Tests/UnitTests/NotificationTests.cs:19]: This video explains how AutoMoq, AutoData actually work: https://www.youtube.com/watch?v=DbK1NVRQZvs&t=2005s

Comment (moltenessence) [reply PMCS/Services/NotificationService/Notification.Tests/UnitTests/NotificationTests.cs:19]: Shorter video by the same author: https://www.youtube.com/watch?v=t3s3-Q1gACk

**Review by Blackolf931 — APPROVED** [review]

**Pull request merged**